kernel/src/hil/sensors.rs

```diff
@@ -233,3 +233,32 @@ pub trait PressureClient {
     /// Returns the value in hPa.
     fn callback(&self, pressure: Result<u32, ErrorCode>);
 }
+
+/// A basic interface for distance sensor.
+pub trait Distance<'a> {
+    /// Set the client
+    fn set_client(&self, client: &'a dyn DistanceClient);
+
+    /// Initiates a distance reading from the sensor.
+    ///
+    /// This function might return the following errors:
+    /// - `BUSY`: Indicates that the hardware is currently busy.
+    /// - `FAIL`: Indicates that there was a failure in communication.
+    /// - `NOSUPPORT`: Indicates that this operation is not supported by the sensor.
```

Comment:
I can see where this came from, but I think we should remove `NOSUPPORT`. If the distance sensor can't do this then this trait shouldn't be implemented.